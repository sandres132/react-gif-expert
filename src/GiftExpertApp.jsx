import React, { useState } from 'react'

import { AddCategory, GifGrid } from './components';

export const GiftExpertApp = () => {

    const [categories, setCategories] = useState(['ONE PIECE ANIME']);

    const onAddCategory = (newCategory) => {
        if(categories.includes(newCategory.toUpperCase())){
            categories.splice(categories.indexOf(newCategory.toUpperCase()),1);
        }
        setCategories([newCategory.toUpperCase(), ...categories ])
    }

  return (
    <>
        <h1>GiftExpertApp</h1>

        <AddCategory 
            onNewCategory={onAddCategory}
        />

        {
            categories.map(category => (
                <GifGrid key={category} cat={category}/>
            ))
        }

    </>
  )
}
