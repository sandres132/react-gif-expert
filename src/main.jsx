import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import { GiftExpertApp } from './GiftExpertApp.jsx'
import './styles.css';

createRoot(document.getElementById('root')).render(
  <StrictMode>
    <GiftExpertApp />
  </StrictMode>,
)
