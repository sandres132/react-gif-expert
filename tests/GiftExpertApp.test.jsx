import React from 'react';

import { fireEvent, render, screen } from '@testing-library/react';

import { GiftExpertApp } from '../src/GiftExpertApp';

describe('Pruebas en <GiftExpertApp/>', () => {
    
    const categories = ['ONE PIECE ANIME', 'NARUTO', 'CALAMARDO'];
    const inputValue = 'NARUTO';
    
    test('Debe de hacer match con el snapshot', () => {
        const {container} = render(<GiftExpertApp/>);
        expect(container).toMatchSnapshot();
        //screen.debug();
    });

    test('Debe de eliminar y repostear la categoria en caso de ya existir en el arreglo', () => {
        render(<GiftExpertApp/>);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, {target: {value: categories[1]}});
        fireEvent.submit(form);
        fireEvent.input(input, {target: {value: categories[2]}});
        fireEvent.submit(form);
        screen.debug();
        fireEvent.input(input, {target: {value: categories[1]}});
        fireEvent.submit(form);
        screen.debug();
        const h4Arrays = screen.getAllByRole('heading', {level: 4}).at(0).innerHTML;
        expect(h4Arrays).toContain(inputValue);

    })
    
})