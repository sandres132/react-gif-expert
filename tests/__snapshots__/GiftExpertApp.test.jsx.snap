// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftExpertApp/> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    GiftExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h4>
    ONE PIECE ANIME
  </h4>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
